root/etc/profile.d/discovery.sh

```diff
@@ -0,0 +1,8 @@
+# /etc/profile.d/discovery.sh - shell extensions for discovery
+
+# Changes to this file should also be reflected in the setup-extension systemd
+# service file, as systemd doesn't source this file
+
+export LD_LIBRARY_PATH=/opt/extension/lib
+export RUBYLIB=/opt/extension/lib/ruby
+export PATH=$PATH:/opt/extension/bin
```

Comment:
We've got two options then - either use a hardcoded value in the systemd unit, or use another service unit to write the environment file (this is possible accorind to the systemd.exec man page you linked:

```
The files listed with this directive will be read shortly before the process is executed (more specifically, after all processes from a previous unit state terminated. This means you can generate these files in one unit state, and read it with this option in the next).
```

I quite like this approach actually, we can have `discovery-setup` which downloads the zip and writes the environment file, and then `discover-host` and `discovery-autostart` (new service to run the autostart file) can depend on `discovery-setup` and read the newly written env file. I'll set that up.